x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Dissect.java

```diff
@@ -10,17 +10,42 @@
 import org.elasticsearch.dissect.DissectParser;
 import org.elasticsearch.xpack.esql.core.expression.Attribute;
 import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.expression.NameId;
+import org.elasticsearch.xpack.esql.core.expression.ReferenceAttribute;
 import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
 import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.parser.ParsingException;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.Set;
 
 public class Dissect extends RegexExtract {
     private final Parser parser;
 
     public record Parser(String pattern, String appendSeparator, DissectParser parser) {
 
+        public List<Attribute> keyAttributes(Source src) {
+            Set<String> referenceKeys = parser.referenceKeys();
+            if (referenceKeys.size() > 0) {
+                throw new ParsingException(
```

Comment:
@luigidellaquila when is this possible, in practical terms? Can you give an example of query?

Reply:
DissectParser can create field _names_ (together with values) at runtime, from data, see https://www.elastic.co/guide/en/elasticsearch/reference/current/dissect-processor.html#dissect-modifier-reference-keys

In ES|QL we don't support it because we need to know the schema at planning time.

We have a test for this as well https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java#L756

Reply:
C.f. https://www.elastic.co/guide/en/elasticsearch/reference/current/dissect-processor.html#dissect-key-modifiers, the `*` and `&` modifiers are disallowed by this validation. (Just checked.) If I understand correctly, this would mean that the output column is named after whatever dissect extracts there, which is invalid for us because our output columns (attributes) need to have predetermined names.